packages/abstractions/src/requestInformation.ts

```diff
@@ -84,18 +84,12 @@ export class RequestInformation {
   public addRequestHeaders(source: Record<string, string[]> | undefined) {
     if (!source) return;
     for (const key in source) {
-      this.headers[key] = source[key];
+      this.headers.add(key, ...source[key]);
     }
   }
   /** Try to add the header for the request if it's not already present. */
   public tryAddRequestHeaders(key: string, value: string): boolean {
```

Comment:
once https://github.com/microsoft/kiota/pull/3678 gets merged, I don't believe we need that method anymore